rust/src/rfb/rfb.rs

```diff
@@ -758,31 +850,6 @@ mod test {
     use super::*;
     use crate::core::STREAM_START;
 
-    #[test]
-    fn test_error_state() {
```

Comment:
I can only imagine this was meant to test the case of invalid data trying to be parsed, ending up in an error state. We can still replicate this case: the only remaining case of this is the first communication (the server announcing its supported protocols) not being correctly parsed -- since the buffer is in no way valid RFB.
I reintroduced this test into the code, and made sure the `parse_response()` is being called, since the server speaks first in this protocol. The test now passes as expected.

Reply:
I wonder more about what becomes of this precise input.

What is the result with this input and the new code ?
A `skip` state and an event ?

Reply:
Yes. I assume this will be one of these cases in which the state machine ends up in an unexpected state.